agent/bench-scripts/pbench-fio

```diff
@@ -588,11 +599,15 @@ function fio_run_job() {
 	echo "running fio job: $fio_job_file ($(basename $benchmark_results_dir))"
 
 	if [ -n "${clients}" ]; then
```

Comment:
That's a really good question. I'd like to have @portante think about this. I *think* that's correct, because the documented behavior is to run locally if there are no clients specified. As is, I suspect we're skipping the `do_local_client` actions in that case, and we probably shouldn't.

I'd love to have confirmation before fixing that and updating more gold files (which currently fail due to the extra debug messages).

Reply:
Yes, this is correct.

The semantics of the `pbench-fio` bench-scripts `--clients` and `--client-file` are defined by fio itself.  By requesting fio separate the client and server portion of fio, we really trying to determine if a `fio --server` command was run remotely or locally.  The `do_local_client` flag is set, it means we have a local `fio --server` process running somewhere and we don't need to use `ssh` to do anything about it.

A possible fix would be to hoist the code at line 658 up to replace line 607.  But I would not advocate we do that in this commit.

See [Section 1.15 "Client/Server"](https://fio.readthedocs.io/en/latest/fio_doc.html#client-server) of the [fio](https://fio.readthedocs.io/en/latest/fio_doc.html) documentation for a more thorough review of the behavior.

Reply:
> Yes, this is correct.

Hmm. I find that a bit ambiguous. 👀 That is, "the original code is correct" or "Webb's concern is correct"??

However, given that without `${clients}` we're presumably running "monolithic" rather than client/server, and the cleanup action (local or remote) involves restarting `fio-server`, I'm assuming you meant (1) ... the code is correct as is, and the cleanup isn't necessary or meaningful without client configuration.

Reply:
> I find that a bit ambiguous.

Me, too!  😆  (I so hate pronouns.)

If the code is correct, perhaps we should have a comment explaining the logic and ramifications (i.e., that, if there are no "clients", then we running "monolithic", and there is nothing else to do)?

Reply:
Dave and I discussed this and we determined that there are some fairly fundamental problems with this script which pre-date this PR and that these changes do not make things materially worse.  Dave is going to open one or more issues describing what we found, and we can take up the discussion in the context of those issues or the PR(s) that they result in.